src/etc/inc/interfaces.lib.inc

```diff
@@ -297,38 +297,36 @@ function legacy_interface_details($intf)
 function configure_interface_hardware($ifs)
 {
     global $config;
-    $intf_details = legacy_interface_details($ifs);
     /* skip vlans for checksumming  */
-    if (!stristr($ifs, "_vlan") && is_array($intf_details)) {
-        // get current settings
-        $csum_set = in_array('rxcsum', $intf_details['options']) || in_array('txcsum', $intf_details['options']);
-        $csumv6_set = in_array('rxcsum6', $intf_details['options']) || in_array('txcsum6', $intf_details['options']);
-        $tso_set = in_array('tso4', $intf_details['options']) || in_array('tso6', $intf_details['options']);
-        $lro_set = in_array('lro', $intf_details['options']);
+    if (!stristr($ifs, "_vlan")) {
 
-        // hardware checksum offloading offloading
-        if (isset($config['system']['disablechecksumoffloading']) && $csum_set) {
-            legacy_interface_flags($ifs, '-txcsum -rxcsum', false);
-        } elseif (!isset($config['system']['disablechecksumoffloading']) && !$csum_set) {
-            legacy_interface_flags($ifs, 'txcsum rxcsum', false);
+        //Get current override settings
+        if(isset($config['system']['offloadoverride'])) {
+            $overrideifs = isset($config['system']['overridemembers']) ? explode(' ', $config['system']['overridemembers']) : array();
+        } else {
+            $overrideifs = array();
         }
-        if (isset($config['system']['disablechecksumoffloading']) && $csumv6_set) {
-            legacy_interface_flags($ifs, '-txcsum6 -rxcsum6', false);
-        } elseif (!isset($config['system']['disablechecksumoffloading']) && !$csumv6_set) {
-            legacy_interface_flags($ifs, 'txcsum6 rxcsum6', false);
+
+	$override = in_array($ifs, $overrideifs);
```

Comment:
I think this merges logic for checkums and tso and lro, we shouldn't want that if we do it right